spec/unit/reports_spec.rb

```diff
@@ -43,6 +43,11 @@
     Puppet::Reports.register_report(:testing) { }
   end
 
+  it "should allow a failed report to be redefined" do
+    expect { Puppet::Reports.register_report(:testing) { raise 'failed report' } }.to raise_error(RuntimeError)
+    Puppet::Reports.register_report(:testing) { }
+  end
+
```

Comment:
This seems like a confusing spec test for two reasons:
1) it's not just *failed* reports that should be able to be reloaded, it's all reports
2) if a report *is* reloaded, we'd expect it to be reloaded the same way

To that end, I'm wondering if the spec test should be something like:
```
  it "should allow a successful report to be reloaded" do
    Puppet::Reports.register_report(:testing) { }
    Puppet::Reports.register_report(:testing) { }
  end

  it "should allow a failed report to be reloaded and show the correct exception both times" do
    expect { Puppet::Reports.register_report(:testing) { raise TypeError, 'failed report' } }.to raise_error(TypeError)
    expect { Puppet::Reports.register_report(:testing) { raise TypeError, 'failed report' } }.to raise_error(TypeError)
  end
```